### ISSUE TITLE
Passing native mongodb connection to mongoose possible?

### ISSUE THREAD

**Issue by hosein** [Automattic/mongoose#9289]
Not a bug report, nor really a feature request, more of a question that I cannot seem to find the answer to:

I have a module that I intend to re-use among several projects that uses the nodejs mongodb driver to manage the various ways my projects connect to various mongodb deployments.  In one deployment I **_need_** to use mongoose, is there any way to hand the connection established by the nodejs mongodb driver to mongoose and skip the mongoose connection methods?  

Pseudocode example:

```javascript
const mongoose = require('mongoose');
const myMongoDBConnector = require('myMongoDBConnector');

myMongoDBConnector.connect({various: options},(err,db)=>{
//Tell mongoose to use this for connection
    mongoose.db = db;
//These are my models as a module
    const models = require('./mongooseModels.js');
//Example query
    models.MyModel.findOne({foor:'bar'},(err,foo)=>{
        console.log(foo);
    }
})
```

**Comment by hosein** [comment]
I've spent so much time trying to find the _**correct**_ way to do this via docs, that I didn't think it could be as stupidly simple as:

```javascript
MongoClient.connect(url,function(err,db){
    mongoose.connection = db;
})

**Comment by vkarpov15** [comment]
Not currently, but a new feature supporting this will be released in Mongoose 5.10 when we release it (we're currently waiting on the MongoDB driver to ship v3.6 before we release 5.10)

```javascript
// Assuming `client` is an instance of `require('mongodb').MongoClient`
mongoose.connection.setClient(client);
```

See #9164
